src/modules/haskell.rs

```diff
@@ -0,0 +1,60 @@
+use ansi_term::Color;
+use std::process::Command;
+
+use super::{Context, Module};
+
+/// Creates a module with the current Haskell Stack version
+///
+/// Will display the Haskell version if any of the following criteria are met:
+///     - Current directory contains a `stack.yaml` file
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let is_haskell_project = context
+        .try_begin_scan()?
+        .set_files(&["stack.yaml"])
+        .is_match();
+
+    if !is_haskell_project {
+        return None;
```

Comment:
there is no need to `return` here or `match` on the haskell version. You can simply say something like
```
if !is_haskell_project {
  None
} else {
  get_haskell_version().map(|haskell_version| {
  ...
  module
})
}
```

Reply:
@bijancn That makes perfect sense. 👍